README.md

```diff
@@ -1,27 +1,305 @@
+Create one in the [Descope Console](https://app.descope.com/settings/company/managementkeys).
+
+```python
+from descope import DescopeClient
+
+# Initialized after setting the DESCOPE_PROJECT_ID and the DESCOPE_MANAGEMENT_KEY env vars
+descope_client = DescopeClient()
+
+# ** Or directely **
+descope_client = DescopeClient(project_id="<Project ID>", management_key="<Management Key>")
+```
+
+### Manage Tenants
+
+You can create, update, delete or load tenants:
+
+```Python
+# You can optionally set your own ID when creating a tenant
+descope_client.mgmt.tenant.create(
+    name="My First Tenant",
+    id="my-custom-id", # This is optional. If omitted
+    self_provisioning_domains=["domain.com"],
+)
+
+# Update will override all fields as is. Use carefully.
+descope_client.mgmt.tenant.update(
+    id="my-custom-id",
+    name="My First Tenant",
+    self_provisioning_domains=["domain.com", "another-domain.com"],
+)
+
+# Tenant deletion cannot be undone. Use carefully.
+descope_client.mgmt.tenant.delete("my-custom-id")
+
+# Load all tenants
+tenants_resp = descope_client.mgmt.tenant.load_all()
+tenants = tenants_resp["tenants"]
+    for tenant in tenants:
+        # Do something
+```
+
+### Manage Users
+
+You can create, update, delete or load users, as well as search according to filters:
+
+```Python
+# A user must have an identifier, other fields are optional.
+# Roles should be set directly if no tenants exist, otherwise set
+# on a per-tenant basis.
+descope_client.mgmt.user.create(
+    identifier="desmond@descope.com",
+    email="desmond@descope.com",
+    display_name="Desmond Copeland",
+    user_tenants=[
+        UserTenant("my-tenant-id", ["role-name1"]),
+    ],
+)
+
+# Update will override all fields as is. Use carefully.
+descope_client.mgmt.user.update(
+    identifier="desmond@descope.com",
+    email="desmond@descope.com",
+    display_name="Desmond Copeland",
+    user_tenants=[
+        UserTenant("my-tenant-id", ["role-name1", "role-name2"]),
+    ],
+)
+
+# Tenant deletion cannot be undone. Use carefully.
+descope_client.mgmt.user.delete("desmond@descope.com")
+
+# Load specific user
+user_resp = descope_client.mgmt.user.load("desmond@descope.com")
+user = user_resp["user"]
+
+# Search all users, optionally according to tenant and/or role filter
+users_resp = descope_client.mgmt.tenant.search_all(tenant_ids=["my-tenant-id"])
+users = users_resp["users"]
+    for user in users:
+        # Do something
+```
+
+### Manage SSO Setting
+
+You can manage SSO settings and map SSO group roles and user attributes.
+
+```Python
+# You can configure SSO settings manually by setting the required fields directly
+descope_client.mgmt.sso.configure(
+    tenant_id, # Which tenant this configuration is for
+    idp_url="https://idp.com",
+    entity_id="my-idp-entity-id",
+    idp_cert="1231232132131", # Send the cert contents only, with the "start prefix" and "end suffix"
```

Comment:
better you use some example with the prefix/suffix, so it will be clear what you are taking about

```
-----BEGIN CERTIFICATE-----
cert...
-----END CERTIFICATE-----
```